packages/cli/src/lib/sandbox/docker.ts

```diff
@@ -0,0 +1,389 @@
+import { getAIAgentTool, getUserAIAgent } from '../agents/index.js';
+import { join } from 'node:path';
+import { AI_AGENT, ProjectConfig } from '../config.js';
+import { Sandbox } from './types.js';
+import { SetupBuilder } from '../setup.js';
+import { TaskDescription } from '../description.js';
+import { findProjectRoot, launch, ProcessManager } from 'rover-common';
+import {
+  parseCustomEnvironmentVariables,
+  loadEnvsFile,
+} from '../../utils/env-variables.js';
+import { existsSync, mkdirSync, mkdtempSync, writeFileSync } from 'node:fs';
+import { homedir, tmpdir, userInfo } from 'node:os';
+import { generateRandomId } from '../../utils/branch-name.js';
+
+const AGENT_IMAGE = 'ghcr.io/endorhq/rover/node:v1.3.1';
+
+export class DockerSandbox extends Sandbox {
+  backend = 'docker';
+
+  constructor(task: TaskDescription, processManager?: ProcessManager) {
+    super(task, processManager);
+  }
+
+  async isBackendAvailable(): Promise<boolean> {
+    try {
+      await launch('docker', ['--version']);
+      return true;
+    } catch (error) {
+      return false;
+    }
+  }
+
+  protected async create(): Promise<string> {
+    // Load task description
+    const roverPath = join(findProjectRoot(), '.rover');
+    const tasksPath = join(roverPath, 'tasks');
+    const taskPath = join(tasksPath, this.task.id.toString());
+    const worktreePath = join(taskPath, 'workspace');
+    const iterationPath = join(
+      taskPath,
+      'iterations',
+      this.task.iterations.toString()
+    );
+    const iterationJsonPath = join(
+      this.task.iterationsPath(),
+      this.task.iterations.toString(),
+      'iteration.json'
+    );
```

Comment:
We should create some helpers on the `IterationConfig` and `Description` classes. I will work on it when migrating them to the `rover-schemas` package